Calculator/ViewController.swift

```diff
@@ -11,102 +11,93 @@ import UIKit
 class ViewController: UIViewController
 {
     @IBOutlet weak var display: UILabel!
+    @IBOutlet weak var history: UILabel!
     
     var userIsInTheMiddleOfTypingANumber = false
     var floatingPointInNumber = false
+    var brain = CalculatorBrain()
     
     @IBAction func appendDigit(sender: UIButton) {
-        let digit = sender.currentTitle!
+        if let digit = sender.currentTitle {
         if userIsInTheMiddleOfTypingANumber {
             display.text = display.text! + digit
+            history.text = history.text! + digit
         } else {
             display.text = digit
+            history.text = history.text! + digit
             userIsInTheMiddleOfTypingANumber = true
         }
+        }
     }
     @IBAction func appendFloatingPoint(sender: UIButton) {
         if floatingPointInNumber {
             display.text = display.text!
+            history.text = history.text!
         } else {
             display.text = display.text! + sender.currentTitle!
+            history.text = history.text! + sender.currentTitle!
             floatingPointInNumber = true
         }
     }
-
+    
     @IBAction func pi(sender: UIButton) {
         let x = M_PI
         if userIsInTheMiddleOfTypingANumber {
             display.text = display.text!
+            history.text = history.text!
         } else {
             display.text = "\(M_PI)"
+            history.text = history.text! + "\(M_PI)"
         }
     }
+    
     @IBAction func operate(sender: UIButton) {
-        let operation = sender.currentTitle!
-        history.append("\(operation)")
+        history.text = history.text! + "\(sender.currentTitle)"
         if userIsInTheMiddleOfTypingANumber {
             enter()
         }
-        switch operation{
-        case "×": performOperation { $0 * $1 }
-        case "÷": performOperation { $1 / $0 }
-        case "+": performOperation { $0 + $1 }
-        case "−": performOperation { $1 - $0 }
-        case "√": performOperation { sqrt($0) }
-        case "sin": performOperation { sin($0) }
-        case "cos": performOperation { cos($0) }
-        default: break
+        if let operation = sender.currentTitle {
+            if let result = brain.performOperation(operation) {
+            displayValue = result
+            }
+         else {
+            displayValue = nil
+        }
         }
     }
     
     @IBAction func clear(sender: UIButton) {
-        display.text = ""
-        operandStack.removeAll(keepCapacity: true)
-        history.removeAll(keepCapacity: true)
-        println(operandStack)
+        display.text = " "
```

Comment:
Remember the purpose of `var displayValue()`. Should rarely see directly manipulating `display.text`